### PR TITLE
fix(pf4): refactor wizard, validating disable nav

### PR DESCRIPTION
**Pull request by rvsia**
- refactors code, make it more simple and cleaner
- fix bugs when there are some fields on the last step
- disabling navigation when validating is in progress
- optimize subscription more

validating

### CODE REVIEW

File: packages/pf4-component-mapper/src/files/wizard.js [hunk 1]
```diff
@@ -89,27 +85,33 @@ const WizardInternal = ({
         {title && <WizardHeader title={title} description={description} onClose={formOptions.onCancel} />}
         <div className="pf-c-wizard__outer-wrap">
           <WizardNav>
-            <FormSpy>
-              {({ values }) => (
+            <FormSpy subscription={{ values: true, valid: true, validating: true }}>
```
Comment (Hyperkid123) [inline packages/pf4-component-mapper/src/files/wizard.js:88]: what about submitting? I think we should block the navigation while submitting as well.

File: packages/pf4-component-mapper/src/files/wizard/wizard-nav.js [hunk 2]
```diff
@@ -20,19 +20,21 @@ const memoValues = (initialValue) => {
...
 
 const WizardNavigationInternal = React.memo(
-  ({ navSchema, activeStepIndex, maxStepIndex, jumpToStep, formOptions: { valid } }) =>
+  ({ navSchema, activeStepIndex, maxStepIndex, jumpToStep, valid, validating }) =>
     navSchema
       .filter((field) => field.primary)
       .map((step) => {
         const substeps = step.substepOf && navSchema.filter((field) => field.substepOf === step.substepOf);
 
+        const isValid = valid && !validating;
```
Comment (Hyperkid123) [inline packages/pf4-component-mapper/src/files/wizard/wizard-nav.js:29]: ^^

File: packages/pf4-component-mapper/src/files/wizard.js (hunk 1 above)
Comment (rvsia) [reply packages/pf4-component-mapper/src/files/wizard.js:88]: the whole navigation? this just disables next steps (on the last step there are no next steps)

Comment (Hyperkid123) [reply packages/pf4-component-mapper/src/files/wizard.js:88]: Is it ok to move in the wizard when the data is in the submission process? I think the navigation in the wizard should be postponed.

Comment (rvsia) [reply packages/pf4-component-mapper/src/files/wizard.js:88]: After submit, the wizard form should go away and be replace by spinner/loading progress according to design specs - there should be no navigation or buttons at all.

**Comment by rvsia** [comment]
+ I am also fixed a little bug in the reducer, when users click on the current step on the initial page (there wasn't any history, so the wizard failed.)

File: packages/pf4-component-mapper/src/files/wizard.js (hunk 1 above)
Comment (Hyperkid123) [reply packages/pf4-component-mapper/src/files/wizard.js:88]: Yes but we are talking about the time period when the submit is in progress not when it was finished.

**Pull request merged**